v23/vom/splice.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 The Vanadium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package vom
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"v.io/v23/vdl"
+)
+
+var (
+	errEncodedBytesNotSupported = errors.New("pre-encoded bytes are not supported by the underlying vom encoder/decoder type")
+)
+
+// MergeEncodedBytes represents an already encoded vom stream that is intended
+// for use when merging the output of multiple, independent, vom encoders.
+type MergeEncodedBytes struct {
+	Data []byte
+}
+
+// VDLWrite implements vdl.Writer.
+func (eb *MergeEncodedBytes) VDLWrite(enc vdl.Encoder) error {
+	uenc, ok := enc.(*encoder81)
+	if !ok {
+		return errEncodedBytesNotSupported
+	}
+	rd := bytes.NewBuffer(eb.Data)
+	dec := NewDecoder(rd)
+	for {
+		var any vdl.Value
+		if err := dec.Decode(&any); err != nil {
```

Comment:
Would be helpful to mention somewhere that this decoding is taking place inside this function.